python/cuml/cluster/hdbscan/prediction.pyx

```diff
@@ -300,6 +307,9 @@ def membership_vector(clusterer, points_to_predict, batch_size=4096, convert_dty
                          "Please call clusterer.fit again with "
                          "prediction_data=True")
 
+    if batch_size <= 0:
+        raise ValueError("batch_size should be in integer that is > 0")
+
```

Comment:
Is this change sufficient to handle the scenario of [1, 4095] points to predict with the default batch size?

Reply:
This will have to be done on the C++ side anyway (the link that @csadorf pointed to) because we need to handle such batch sizes for users of the C++ API. So I did not want to do a `min(... )` in both places.

Reply:
Doing this in C++ makes sense. I only wanted to flag this initially as the example in the linked issue had a batch_size of 4096 (default) with 100 points to predict and still threw the error. Would we expect that to succeed currently?